python/hail/representation/genotype.py

```diff
@@ -0,0 +1,297 @@
+
+    def is_not_called(self):
+        """True if the genotype call is missing.
+
+        :rtype: bool
+        """
+
+        return self._jrep.isNotCalled()
+
+    def is_called(self):
+        """True if the genotype call is non-missing.
+
+        :rtype: bool
+        """
+
+        return self._jrep.isCalled()
+
+    def num_alt_alleles(self):
+        """Returns the count of non-reference alleles.
+
+        This function returns None if the genotype call is missing.
+
+        :rtype: int or None
+        """
+
+        return from_option(self._jrep.nNonRefAlleles())
+
+    def one_hot_alleles(self, num_alleles):
+        """Returns a list containing the one-hot encoded representation of the called alleles.
+
+        This one-hot representation is the positional sum of the one-hot
+        encoding for each called allele.  For a biallelic variant, the
+        one-hot encoding for a reference allele is [1, 0] and the one-hot
+        encoding for an alternate allele is [0, 1].  Thus, with the
+        following variables:
+
+        >>> num_alleles = 2
+        >>> hom_ref = Genotype(0)
+        >>> het = Genotype(1)
+        >>> hom_var = Genotype(2)
+
+        All the below statements are true:
+
+        >>> hom_ref.one_hot_alleles(num_alleles) == [2, 0]
+        >>> het.one_hot_alleles(num_alleles) == [1, 1]
+        >>> hom_var.one_hot_alleles(num_alleles) == [0, 2]
+
+        This function returns None if the genotype call is missing.
+
+        :param int num_alleles: number of possible alternate alleles
+        :rtype: list of int or None
+        """
+
+        result = from_option(self._jrep.oneHotAlleles(num_alleles))
+        if result:
+            result = [x for x in scala_package_object(Env.hail_package().utils).iterableToArrayList(result)]
+        return result
+
+    def one_hot_genotype(self, num_genotypes):
+        """Returns a list containing the one-hot encoded representation of the genotype call.
+
+        A one-hot encoding is a vector with one '1' and many '0' values, like
+        [0, 0, 1, 0] or [1, 0, 0, 0].  This function is useful for transforming
+        the genotype call (gt) into a one-hot encoded array.  With the following
+        variables:
+
+        >>> num_genotypes = 3
+        >>> hom_ref = Genotype(0)
+        >>> het = Genotype(1)
+        >>> hom_var = Genotype(2)
+
+        All the below statements are true:
+
+        >>> hom_ref.one_hot_genotype(num_genotypes) == [1, 0, 0]
+        >>> het.one_hot_genotype(num_genotypes) == [0, 1, 0]
+        >>> hom_var.one_hot_genotype(num_genotypes) == [0, 0, 1]
+
+        This function returns None if the genotype call is missing.
+
+        :param int num_genotypes: number of possible genotypes
+        :rtype: list of int or None
+        """
+
+        result = from_option(self._jrep.oneHotGenotype(num_genotypes))
+        if result:
+            result = [x for x in scala_package_object(Env.hail_package().utils).iterableToArrayList(result)]
```

Comment:
You might make iterableToArrayList into into jiterable_to_list convenience function that handles nulls/None in the java package.